src/internal/MonoColor.cpp

```diff
@@ -0,0 +1,124 @@
+/*-------------------------------------------------------------------------
+MonoColor provides a color object that can be directly consumed by NeoPixelBus
+
+Written by Grzegorz Bialokziewicz (based on Michael C. Miller code).
+
+I invest time and resources providing this open source code,
+please support me by dontating (see https://github.com/Makuna/NeoPixelBus)
+
+-------------------------------------------------------------------------
+This file is part of the Makuna/NeoPixelBus library.
+
+NeoPixelBus is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+NeoPixelBus is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+
+#include "MonoColor.h"
+#include "RgbColor.h"
+#include "Rgb48Color.h"
+#include "HslColor.h"
+#include "HsbColor.h"
+#include "HtmlColor.h"
+
+MonoColor::MonoColor(uint8_t r, uint8_t g, uint8_t b, uint8_t w)
+{
+    W = max(static_cast<uint8_t>(r * 0.3 + g * 0.59 + b * 0.11), w);
+}
+
+MonoColor::MonoColor(uint8_t brightness)
+{
+    W = brightness;
+}
+
+MonoColor::MonoColor(const RgbColor& color)
+{
+    W = color.R * 0.3 + color.G * 0.59 + color.B * 0.11;
+}
+
+MonoColor::MonoColor(const HtmlColor& color)
+{
+    uint32_t temp = color.Color;
+    W = max(static_cast<uint8_t>(((temp >> 16) & 0xff) * 0.3 + ((temp >> 8) & 0xff) * 0.59 + (temp & 0xff) * 0.11), static_cast<uint8_t>((temp >> 24) & 0xff));
+};
+
+MonoColor::MonoColor(const HslColor& color)
+{
+    W = color.L * Max;
```

Comment:
I don't believe this is accurate.
[HslColor Wiki Entry](https://github.com/Makuna/NeoPixelBus/wiki/Color-objects#hslcolor-object)
Saturation has an effect on "whiteness" of the colors, so generally L property should be limited between 0.0 and 0.5.  But testing is required.